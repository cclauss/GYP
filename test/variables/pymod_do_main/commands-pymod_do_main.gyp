{
  'variables': {
    'names': [
      "arguments",
      "array",
      "array-copywithin",
      "array-filter",
      "array-find",
      "array-findindex",
      "array-foreach",
      "array-join",
      "array-map",
      "array-of",
      "array-reverse",
      "array-slice",
      "array-splice",
      "array-unshift",
      "array-lastindexof",
      "base",
      "collections",
      "data-view",
      "extras-utils",
      "growable-fixed-array",
      "iterator",
      "object",
      "string",
      "typed-array",
      "typed-array-createtypedarray",
      "typed-array-filter",
      "typed-array-foreach",
      "typed-array-reduce",
      "typed-array-reduceright",
      "typed-array-slice",
      "typed-array-subarray",
    ],
    'output_root': 'real-output-root',
    'ret1': [
      '<!@pymod_do_main(ForEachFormat "<(output_root)/%s-from-name.cc" <@(names))',
      '<!@pymod_do_main(ForEachFormat "<(output_root)/%s-from-name.hpp" <@(names))'
    ],
    'ret1+': [ '<!@pymod_do_main(ForEachFormat "<(output_root)/%s-from-name.h" <@(names))' ],
  },
  'includes': [
    'commands-pymod_do_main.gypi',
  ],
  'targets': [
    {
      'target_name': 'foo',
      'type': 'none',
      'variables': {
        'ret2': '<!pymod_do_main(GetConstantValue 2)',
      },
      # 'actions': [
      #   {
      #     'action_name': 'test_action',
      #     'variables': {
      #       'ret3': '<!pymod_do_main(GetConstantValue 3)',
      #     },
      #     'inputs' : [
      #       '<@(ret1)',
      #     ],
      #     'outputs': [
      #       '<(ret2)',
      #       '<(ret3)',
      #       '<(ret4)',
      #     ],
      #     'action': [
      #       'echo',
      #       '<@(_inputs)',
      #       '<@(_outputs)',
      #     ],
      #   },
      # ],
    },
  ],
}
