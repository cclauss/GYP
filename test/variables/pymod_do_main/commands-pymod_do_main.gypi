# This file is included from commands-pymod_do_main.gyp to test evaluation order of includes.
{
  'variables': {
    'ret4': '<!pymod_do_main(GetConstantValue 4)',
  },
  'targets': [
    {
      'target_name': 'dummy',
      'type': 'none',
      'variables': {
        'ret5': '<!pymod_do_main(GetConstantValue 5)',
      },
    },
  ],
}
