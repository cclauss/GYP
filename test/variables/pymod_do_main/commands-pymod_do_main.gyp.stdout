VARIABLES:input.py:562:ExpandVariables Matches: {'replace': '<!pymod_do_main(GetConstantValue 4)', 'type': '<!', 'command_string': 'pymod_do_main', 'is_array': '', 'content': 'GetConstantValue 4'}
VARIABLES:input.py:684:ExpandVariables Executing command 'GetConstantValue 4' in directory 'None'
VARIABLES:input.py:798:ExpandVariables Found output 'Lorem ipsum. 4', recursing.
VARIABLES:input.py:562:ExpandVariables Matches: {'replace': '<@(names)', 'type': '<@', 'command_string': None, 'is_array': '', 'content': 'names'}
VARIABLES:input.py:562:ExpandVariables Matches: {'replace': '<!@pymod_do_main(ForEachFormat "<(output_root)', 'type': '<!@', 'command_string': 'pymod_do_main', 'is_array': '', 'content': 'ForEachFormat "<(output_root'}
VARIABLES:input.py:562:ExpandVariables Matches: {'replace': '<(output_root)', 'type': '<', 'command_string': None, 'is_array': '', 'content': 'output_root'}
VARIABLES:input.py:798:ExpandVariables Found output 'ForEachFormat "real-output-root/%s-from-name.cc" arguments array array-copywithin array-filter array-find array-findindex array-foreach array-join array-map array-of array-reverse array-slice array-splice array-unshift array-lastindexof base collections data-view extras-utils growable-fixed-array iterator object string typed-array typed-array-createtypedarray typed-array-filter typed-array-foreach typed-array-reduce typed-array-reduceright typed-array-slice typed-array-subarray', recursing.
VARIABLES:input.py:684:ExpandVariables Executing command 'ForEachFormat "real-output-root/%s-from-name.cc" arguments array array-copywithin array-filter array-find array-findindex array-foreach array-join array-map array-of array-reverse array-slice array-splice array-unshift array-lastindexof base collections data-view extras-utils growable-fixed-array iterator object string typed-array typed-array-createtypedarray typed-array-filter typed-array-foreach typed-array-reduce typed-array-reduceright typed-array-slice typed-array-subarray' in directory 'None'
VARIABLES:input.py:798:ExpandVariables Found output ['real-output-root/arguments-from-name.cc', 'real-output-root/array-from-name.cc', 'real-output-root/array-copywithin-from-name.cc', 'real-output-root/array-filter-from-name.cc', 'real-output-root/array-find-from-name.cc', 'real-output-root/array-findindex-from-name.cc', 'real-output-root/array-foreach-from-name.cc', 'real-output-root/array-join-from-name.cc', 'real-output-root/array-map-from-name.cc', 'real-output-root/array-of-from-name.cc', 'real-output-root/array-reverse-from-name.cc', 'real-output-root/array-slice-from-name.cc', 'real-output-root/array-splice-from-name.cc', 'real-output-root/array-unshift-from-name.cc', 'real-output-root/array-lastindexof-from-name.cc', 'real-output-root/base-from-name.cc', 'real-output-root/collections-from-name.cc', 'real-output-root/data-view-from-name.cc', 'real-output-root/extras-utils-from-name.cc', 'real-output-root/growable-fixed-array-from-name.cc', 'real-output-root/iterator-from-name.cc', 'real-output-root/object-from-name.cc', 'real-output-root/string-from-name.cc', 'real-output-root/typed-array-from-name.cc', 'real-output-root/typed-array-createtypedarray-from-name.cc', 'real-output-root/typed-array-filter-from-name.cc', 'real-output-root/typed-array-foreach-from-name.cc', 'real-output-root/typed-array-reduce-from-name.cc', 'real-output-root/typed-array-reduceright-from-name.cc', 'real-output-root/typed-array-slice-from-name.cc', 'real-output-root/typed-array-subarray-from-name.cc'], recursing.
VARIABLES:input.py:562:ExpandVariables Matches: {'replace': '<@(names)', 'type': '<@', 'command_string': None, 'is_array': '', 'content': 'names'}
VARIABLES:input.py:562:ExpandVariables Matches: {'replace': '<!@pymod_do_main(ForEachFormat "<(output_root)', 'type': '<!@', 'command_string': 'pymod_do_main', 'is_array': '', 'content': 'ForEachFormat "<(output_root'}
VARIABLES:input.py:562:ExpandVariables Matches: {'replace': '<(output_root)', 'type': '<', 'command_string': None, 'is_array': '', 'content': 'output_root'}
VARIABLES:input.py:798:ExpandVariables Found output 'ForEachFormat "real-output-root/%s-from-name.hpp" arguments array array-copywithin array-filter array-find array-findindex array-foreach array-join array-map array-of array-reverse array-slice array-splice array-unshift array-lastindexof base collections data-view extras-utils growable-fixed-array iterator object string typed-array typed-array-createtypedarray typed-array-filter typed-array-foreach typed-array-reduce typed-array-reduceright typed-array-slice typed-array-subarray', recursing.
VARIABLES:input.py:684:ExpandVariables Executing command 'ForEachFormat "real-output-root/%s-from-name.hpp" arguments array array-copywithin array-filter array-find array-findindex array-foreach array-join array-map array-of array-reverse array-slice array-splice array-unshift array-lastindexof base collections data-view extras-utils growable-fixed-array iterator object string typed-array typed-array-createtypedarray typed-array-filter typed-array-foreach typed-array-reduce typed-array-reduceright typed-array-slice typed-array-subarray' in directory 'None'
VARIABLES:input.py:798:ExpandVariables Found output ['real-output-root/arguments-from-name.hpp', 'real-output-root/array-from-name.hpp', 'real-output-root/array-copywithin-from-name.hpp', 'real-output-root/array-filter-from-name.hpp', 'real-output-root/array-find-from-name.hpp', 'real-output-root/array-findindex-from-name.hpp', 'real-output-root/array-foreach-from-name.hpp', 'real-output-root/array-join-from-name.hpp', 'real-output-root/array-map-from-name.hpp', 'real-output-root/array-of-from-name.hpp', 'real-output-root/array-reverse-from-name.hpp', 'real-output-root/array-slice-from-name.hpp', 'real-output-root/array-splice-from-name.hpp', 'real-output-root/array-unshift-from-name.hpp', 'real-output-root/array-lastindexof-from-name.hpp', 'real-output-root/base-from-name.hpp', 'real-output-root/collections-from-name.hpp', 'real-output-root/data-view-from-name.hpp', 'real-output-root/extras-utils-from-name.hpp', 'real-output-root/growable-fixed-array-from-name.hpp', 'real-output-root/iterator-from-name.hpp', 'real-output-root/object-from-name.hpp', 'real-output-root/string-from-name.hpp', 'real-output-root/typed-array-from-name.hpp', 'real-output-root/typed-array-createtypedarray-from-name.hpp', 'real-output-root/typed-array-filter-from-name.hpp', 'real-output-root/typed-array-foreach-from-name.hpp', 'real-output-root/typed-array-reduce-from-name.hpp', 'real-output-root/typed-array-reduceright-from-name.hpp', 'real-output-root/typed-array-slice-from-name.hpp', 'real-output-root/typed-array-subarray-from-name.hpp'], recursing.
VARIABLES:input.py:562:ExpandVariables Matches: {'replace': '<@(names)', 'type': '<@', 'command_string': None, 'is_array': '', 'content': 'names'}
VARIABLES:input.py:562:ExpandVariables Matches: {'replace': '<!@pymod_do_main(ForEachFormat "<(output_root)', 'type': '<!@', 'command_string': 'pymod_do_main', 'is_array': '', 'content': 'ForEachFormat "<(output_root'}
VARIABLES:input.py:562:ExpandVariables Matches: {'replace': '<(output_root)', 'type': '<', 'command_string': None, 'is_array': '', 'content': 'output_root'}
VARIABLES:input.py:798:ExpandVariables Found output 'ForEachFormat "real-output-root/%s-from-name.h" arguments array array-copywithin array-filter array-find array-findindex array-foreach array-join array-map array-of array-reverse array-slice array-splice array-unshift array-lastindexof base collections data-view extras-utils growable-fixed-array iterator object string typed-array typed-array-createtypedarray typed-array-filter typed-array-foreach typed-array-reduce typed-array-reduceright typed-array-slice typed-array-subarray', recursing.
VARIABLES:input.py:684:ExpandVariables Executing command 'ForEachFormat "real-output-root/%s-from-name.h" arguments array array-copywithin array-filter array-find array-findindex array-foreach array-join array-map array-of array-reverse array-slice array-splice array-unshift array-lastindexof base collections data-view extras-utils growable-fixed-array iterator object string typed-array typed-array-createtypedarray typed-array-filter typed-array-foreach typed-array-reduce typed-array-reduceright typed-array-slice typed-array-subarray' in directory 'None'
VARIABLES:input.py:798:ExpandVariables Found output ['real-output-root/arguments-from-name.h', 'real-output-root/array-from-name.h', 'real-output-root/array-copywithin-from-name.h', 'real-output-root/array-filter-from-name.h', 'real-output-root/array-find-from-name.h', 'real-output-root/array-findindex-from-name.h', 'real-output-root/array-foreach-from-name.h', 'real-output-root/array-join-from-name.h', 'real-output-root/array-map-from-name.h', 'real-output-root/array-of-from-name.h', 'real-output-root/array-reverse-from-name.h', 'real-output-root/array-slice-from-name.h', 'real-output-root/array-splice-from-name.h', 'real-output-root/array-unshift-from-name.h', 'real-output-root/array-lastindexof-from-name.h', 'real-output-root/base-from-name.h', 'real-output-root/collections-from-name.h', 'real-output-root/data-view-from-name.h', 'real-output-root/extras-utils-from-name.h', 'real-output-root/growable-fixed-array-from-name.h', 'real-output-root/iterator-from-name.h', 'real-output-root/object-from-name.h', 'real-output-root/string-from-name.h', 'real-output-root/typed-array-from-name.h', 'real-output-root/typed-array-createtypedarray-from-name.h', 'real-output-root/typed-array-filter-from-name.h', 'real-output-root/typed-array-foreach-from-name.h', 'real-output-root/typed-array-reduce-from-name.h', 'real-output-root/typed-array-reduceright-from-name.h', 'real-output-root/typed-array-slice-from-name.h', 'real-output-root/typed-array-subarray-from-name.h'], recursing.
VARIABLES:input.py:562:ExpandVariables Matches: {'replace': '<!pymod_do_main(GetConstantValue 2)', 'type': '<!', 'command_string': 'pymod_do_main', 'is_array': '', 'content': 'GetConstantValue 2'}
VARIABLES:input.py:684:ExpandVariables Executing command 'GetConstantValue 2' in directory 'None'
VARIABLES:input.py:798:ExpandVariables Found output 'Lorem ipsum. 2', recursing.
VARIABLES:input.py:562:ExpandVariables Matches: {'replace': '<!pymod_do_main(GetConstantValue 5)', 'type': '<!', 'command_string': 'pymod_do_main', 'is_array': '', 'content': 'GetConstantValue 5'}
VARIABLES:input.py:684:ExpandVariables Executing command 'GetConstantValue 5' in directory 'None'
VARIABLES:input.py:798:ExpandVariables Found output 'Lorem ipsum. 5', recursing.
